lib/wrappers/genesisProtocol.ts

```diff
@@ -332,13 +333,69 @@ export class GenesisProtocolWrapper extends IntVoteInterfaceWrapper
+   * The computation depends on the current number of boosted proposals that were created for
+   * this voting machine by the scheme at `schemeInfo.address` for the given `creatorAddress`
+   * (`creatorAddress` is the `organizationAddress` given to the `propose` method),
+   * and the GenesisProtocol parameters `thresholdConstA` and `thresholdConstB` which
+   * are registered on behalf of the given scheme with an avatar's Controller,
+   * keyed by the given `creatorAddress`.
+   *
+   * Thus `creatorAddress` must be an avatar address and this function will not work if
+   * you pass for `creatorAddress` any other contract address than an avatar.
+   * Otherwise you must use `getThresholdFromProposal`.
+   *
+   * Note that there will be a separately-scoped threshold for proposals created for this
+   * voting machine by any other scheme than the one given.
+   *
+   * Note that for Arc.js schemes you can just pass an Arc.js contract wrapper for `schemeInfo`.
+   */
+  public async getThresholdForSchemeAndCreator(
+    schemeInfo: ThresholdSchemeInfo,
+    creatorAddress: Address): Promise<BigNumber> {
+
+    if (!schemeInfo) {
+      throw new Error("schemeAddress is not defined");
+    }
+
+    if (!creatorAddress) {
+      throw new Error("creatorAddress is not defined");
+    }
+
+    const organizationId = this.organizationIdFromProposalCreator(schemeInfo.address, creatorAddress);
+
+    const votingMachineInfo = await schemeInfo.getSchemeParameters(creatorAddress);
+
+    if (!votingMachineInfo) {
+      throw new Error("votingMachineInfo is null, creatorAddress may not be an avatar");
+    }
+
+    if (votingMachineInfo.votingMachineAddress !== this.address) {
+      throw new Error("the scheme is not using this voting machine");
+    }
+
+    const votingMachineParamsHash = votingMachineInfo.voteParametersHash;
+
+    if (UtilsInternal.isNullHash(votingMachineParamsHash)) {
+      throw new Error("creatorAddress does not yield a parameters hash, may not be an avatar");
+    }
+
+    this.logContractFunctionCall("GenesisProtocol.threshold",
+      { schemeInfo, creatorAddress, votingMachineParamsHash, organizationId });
+
+    return this.contract.threshold(votingMachineParamsHash, organizationId);
+  }
+  /**
+   * Return the threshold that is required for a proposal to be shifted into the boosted state.
+   * The computation depends on the current number of boosted proposals that were created for
+   * this voting machine by the scheme that created the given proposal,
+   * and the GenesisProtocol parameters `thresholdConstA` and `thresholdConstB` which
+   * are stored with the proposal.
+   *
+   * Note that there will be a separately-scoped threshold for proposals created for this
+   * voting machine by any other scheme than the one that created the given proposal.
    */
-  public async getThreshold(proposalId: Hash): Promise<BigNumber> {
+  public async getThresholdFromProposal(proposalId: Hash): Promise<BigNumber> {
```

Comment:
"From" seems wrong here too, isn't it "for"?

Reply:
but I realize that it must in this case be an avatar address (as documented), so I'll change that

Reply:
Done

Reply:
ah interesting and a bit strange but ok